clang/test/SemaSYCL/sycl-kernel-entry-point-attr-argument.cpp

```diff
@@ -209,8 +209,40 @@ struct B25_2;
   clang::sycl_kernel_entry_point(B25_2)]]
 void bad25();
 
-struct B26;
-// expected-error@+3 {{'sycl_kernel_entry_point' kernel name argument conflicts with a previous declaration}}
+struct B26 { void operator()() const {} };
+// expected-error@+3 {{'sycl_kernel_entry_point' kernel name 'B26' conflicts with a previous declaration}}
 // expected-note@+1  {{previous declaration is here}}
-[[clang::sycl_kernel_entry_point(B26)]] void bad26_1();
-[[clang::sycl_kernel_entry_point(B26)]] void bad24_2();
+[[clang::sycl_kernel_entry_point(B26)]] void bad26_1(B26 k) { k(); }
+[[clang::sycl_kernel_entry_point(B26)]] void bad24_2(B26 k) { k(); }
```

Comment:
This test was intentional in exercising declarations that are not definitions. I think we should detect and diagnose such cases (though I do expect non-defining declarations to be rare in practice).

The name `bad24_2` is a pre-existing copy/paste error on my part. That name should be `bad26_2`.

Reply:
> though I do expect non-defining declarations to be rare in practice

I failed to see the use, hence why I happily did that change.

Reply:
Thanks. I failed to document why the test was written that way. Perhaps add a comment along these lines?
```
// Validate that conflicting kernel names are diagnosed for non-defining declarations.
```

Reply:
I addressed this in the new patch